tests/chainer_tests/links_tests/model_tests/test_classifier.py

```diff
@@ -0,0 +1,83 @@
+import unittest
+
+import mock
+import numpy
+
+import chainer
+from chainer import links
+from chainer import testing
+from chainer.testing import attr
+
+
+@testing.parameterize(*testing.product({
+    'compute_accuracy': [True, False],
+    'x_num': [1, 2]
+}))
+class TestClassifier(unittest.TestCase):
+
+    def setUp(self):
+        self.link = links.Classifier(links.Linear(10, 3))
```

Comment:
If we pass `MagicMock` object as the first argument of `Classifier`, it should raises error like this.

```
ERROR: test_call_cpu (test_classifier.transplant_class.<locals>.C)  parameter: {'x_num': 1, 'compute_accuracy': True}
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/delta/dev/chainer2/tests/chainer_tests/links_tests/model_tests/test_classifier.py", line 19, in setUp
    self.link = links.Classifier(mock.MagicMock())
  File "/home/delta/dev/chainer2/chainer/links/model/classifier.py", line 33, in __init__
    super(Classifier, self).__init__(predictor=predictor)
  File "/home/delta/dev/chainer2/chainer/link.py", line 419, in __init__
    self.add_link(name, link)
  File "/home/delta/dev/chainer2/chainer/link.py", line 446, in add_link
    % link.name)
ValueError: given link is already registered to another chain by name <MagicMock name='mock.name' id='140331554906352'>
```

As we can see [here]( https://github.com/pfnet/chainer/blob/master/chainer/link.py#L444), `name` attribute of the `Link` object to be added as a child of some `Chain` should be `None`. But `MagicMock` object returns `MagicMock` object when we access its attribute. 

Of course there may be some way to pass `MagicMock` to `Classifier`. But this workaround is easier.